### PR TITLE
test: Added E2E tests for Divider component

### PR DESCRIPTION
**Pull request by 2020Deeya**
# Description

1] Added E2E tests for Divider component
2] Minor fix in Divider Component 

## Links

https://jira-eng-gpk2.cisco.com/jira/browse/MOMENTUM-434

### CODE REVIEW

File: packages/components/src/components/divider/divider.e2e-test.ts [hunk 1]
```diff
@@ -1,65 +1,208 @@
...
+
+    const dividerStickerSheet = new StickerSheet(componentsPage, 'mdc-divider');
+    await test.step('Horizontal', async () => {
+      // for primary type
+      dividerStickerSheet.setAttributes({ orientation: DIVIDER_ORIENTATION.HORIZONTAL });
+      dividerStickerSheet.setChildren(NoChildren);
+      await dividerStickerSheet.createMarkupWithCombination({ variant: DIVIDER_VARIANT });
+
+      // for text type
+      dividerStickerSheet.setAttributes({ orientation: DIVIDER_ORIENTATION.HORIZONTAL });
```
Comment (supminn) [inline packages/components/src/components/divider/divider.e2e-test.ts:127]: this line can be removed as we are already setting the orientation on line 122.

File: packages/components/src/components/divider/divider.e2e-test.ts [hunk 2]
```diff
@@ -1,65 +1,208 @@
...
+      dividerStickerSheet.setChildren(Text);
+      await dividerStickerSheet.createMarkupWithCombination({ variant: DIVIDER_VARIANT });
 
-    await test.step('focus', async () => {
-      await test.step('component should be focusable with tab', async () => {
-        // TODO: add test here
+      // for grabber type
+      dividerStickerSheet.setAttributes({ orientation: DIVIDER_ORIENTATION.HORIZONTAL });
```
Comment (supminn) [inline packages/components/src/components/divider/divider.e2e-test.ts:132]: same as above comment, this line can be omitted as the orientation is already set on line 122

File: packages/components/src/components/divider/divider.e2e-test.ts [hunk 3]
```diff
@@ -1,65 +1,208 @@
...
-      await test.step('component should fire callback x when pressing y', async () => {
-        // TODO: add test here
+    await test.step('Vertical', async () => {
+      // for primary type
+      dividerStickerSheet.setAttributes({ orientation: DIVIDER_ORIENTATION.VERTICAL });
+      dividerStickerSheet.setChildren(NoChildren);
+      await dividerStickerSheet.createMarkupWithCombination({ variant: DIVIDER_VARIANT });
+
+      // for grabber type
+      dividerStickerSheet.setAttributes({ orientation: DIVIDER_ORIENTATION.VERTICAL });
```
Comment (supminn) [inline packages/components/src/components/divider/divider.e2e-test.ts:172]: this line can be removed as we are already setting the orientation on line 167.

File: packages/components/src/components/divider/divider.e2e-test.ts [hunk 4]
```diff
@@ -1,65 +1,208 @@
...
+      });
+
+      await container.evaluate((el) => {
+        const rowWrappers = el.querySelectorAll('.componentRowWrapper') as NodeListOf<HTMLElement>;
+        Array.from(rowWrappers).forEach((rowWrapper) => {
+          const modifiedRowWrapper = rowWrapper;
+          modifiedRowWrapper.style.gap = '3.5rem';
+          modifiedRowWrapper.style.marginBottom = '3.5rem';
+        });
+      });
```
Comment (supminn) [inline packages/components/src/components/divider/divider.e2e-test.ts:196]:
```suggestion
  await container.evaluate((wrapper) => {
        const modifiedWrapper = wrapper;
        modifiedWrapper.style.display = 'flex';
        modifiedWrapper.style.gap = '3.5rem';
        modifiedWrapper.style.height = '90%';
        modifiedWrapper.style.padding = '1.25rem';

        const rowWrappers = wrapper.querySelectorAll('.componentRowWrapper') as NodeListOf<HTMLElement>;
        Array.from(rowWrappers).forEach((rowWrapper) => {
          const modifiedRowWrapper = rowWrapper;
          modifiedRowWrapper.style.gap = '3.5rem';
          modifiedRowWrapper.style.marginBottom = '3.5rem';
        });
      });
```
Both of them can be combined in a single block of evaluate.

**Review by supminn — CHANGES_REQUESTED** [review]

File: packages/components/src/components/divider/__screenshots__/chrome/mdc-divider-horizontal-high-contrast.png
[the remark this answers is not in the document]
Comment (2020Deeya) [reply packages/components/src/components/divider/__screenshots__/chrome/mdc-divider-horizontal-high-contrast.png:1]: It's an existing issue in Bullet too. We are using background color token where the grey is too light and the browser turns it to white.

File: packages/components/src/components/divider/__screenshots__/chrome/mdc-divider-horizontal-ltr.png
[the remark this answers is not in the document]
Comment (2020Deeya) [reply packages/components/src/components/divider/__screenshots__/chrome/mdc-divider-horizontal-ltr.png:1]: Their variant is not same. That's the difference.

[the remark this answers is not in the document]
Comment (supminn) [reply packages/components/src/components/divider/__screenshots__/chrome/mdc-divider-horizontal-ltr.png:1]: oops, my bad. Thanks for the hightlight.

**Review by supminn — APPROVED** [review]

**Pull request merged**